src/actions/attack.cpp

```diff
@@ -158,7 +158,7 @@ battle_context_unit_stats::battle_context_unit_stats(nonempty_unit_const_ptr up,
 	}
 
 	// Handle plague.
-	unit_ability_list plague_specials = weapon->get_specials_and_abilities("plague");
+	unit_ability_list plague_specials = weapon->get_specials_and_abilities_with_priority("plague", {"Soulless", "Walking Corpse"});
```

Comment:
I haven't understood what's happening here, but hardcoding unit types into the engine is a no-no. UMC authors have to be able to make custom units by copying WML and Lua from mainline units, without needing to change the C++.

Reply:
changes cancelled